lib/modulesync/repository.rb

```diff
@@ -63,22 +63,24 @@ def switch_branch(branch)
       end
     end
 
+    def cloned?
+      Dir.exist? File.join(@directory, '.git')
+    end
+
+    def clone
+      puts "Cloning from '#{@remote}'"
+      @git = Git.clone(@remote, @directory)
+    end
+
     def prepare_workspace(branch)
-      # Repo already cloned, check out master and override local changes
-      if Dir.exist? File.join(@directory, '.git')
-        # Some versions of git can't properly handle managing a repo from outside the repo directory
-        Dir.chdir(@directory) do
```

Comment:
The chdir is no longer needed? I tried looking at the ruby-git code but never looked deep enough to figure out if the bug had been resolved.

Reply:
Never faced myself to the bug, but this line is now old, and AFAICR there is another line that do it with `chdir`

Reply:
And tests are two…

Reply:
I'd be ok with that.